### PR TITLE
Michael

### PR DESCRIPTION
**Pull request by MethodicalMadness**
Methods to be reviewed:
Hotel.book()
Hotel.checkin()
Hotel.addServiceCharge()
Hotel.Checkout()
BookingCTL.creditDetailsEntered()
Testing classes to be reviewed:
HotelTest.java
BookingCTLTest.java

### CODE REVIEW

File: src/hotel/entities/Hotel.java [hunk 1]
```diff
@@ -1,111 +1,134 @@
...
+        return bookingsByConfirmationNumber.get(confirmationNumber);
+    }
+
+
+    public long book(Room room, Guest guest, Date arrivalDate,
+                     int stayLength, int occupantNumber, CreditCard creditCard) {
+        Booking booking =  room.book(guest, arrivalDate, stayLength, occupantNumber, creditCard);
+        long confirmationNumber = booking.getConfirmationNumber();
+        bookingsByConfirmationNumber.put(confirmationNumber, booking);
+        return Long.valueOf(confirmationNumber);
```
Comment (johngalvin09) [inline src/hotel/entities/Hotel.java]: Line 97: the value of ConfirmationNumber is already a long, you can remove the Long.ValueOf(confirmationNumber)and just return confirmationNumber

File: src/hotel/booking/BookingCTL.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
 public class BookingCTL {
 	
 	
-	private static enum State {PHONE, ROOM, REGISTER, TIMES, CREDIT, APPROVED, CANCELLED, COMPLETED}	
+	public static enum State {PHONE, ROOM, REGISTER, TIMES, CREDIT, APPROVED, CANCELLED, COMPLETED}
```
Comment (johngalvin09) [inline src/hotel/booking/BookingCTL.java]: enum State can be set back to private - post testing.

File: src/hotel/entities/Hotel.java (hunk 1 above)
Comment (MethodicalMadness) [reply src/hotel/entities/Hotel.java]: resolved

File: src/hotel/booking/BookingCTL.java (hunk 2 above)
Comment (MethodicalMadness) [reply src/hotel/booking/BookingCTL.java]: resolved

**Review by johngalvin09 — APPROVED** [review]

**Comment by johngalvin09** [comment]
I approve all changes. I am happy with this.

**Comment by MethodicalMadness** [comment]
Facilitator: Michael Coleman
Confirmed that code compiles and appears to work after all implemented code was added.

**Pull request merged**